filters/auth/jwt_validation_test.go

```diff
@@ -0,0 +1,177 @@
+package auth
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/base64"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/golang-jwt/jwt"
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/filters"
+	"github.com/zalando/skipper/net"
+	"github.com/zalando/skipper/proxy/proxytest"
+)
+
+const (
+	kid = "mykid"
+)
+
+var (
+	privateKey, _ = rsa.GenerateKey(rand.Reader, 2048)
+)
+
+func createToken(t *testing.T) string {
+	// Create the Claims
+	claims := &jwt.StandardClaims{
+		ExpiresAt: time.Now().Unix() + 1000,
+		Issuer:    "test",
+		Subject:   "aaa",
+	}
+
+	token := jwt.NewWithClaims(jwt.SigningMethodRS256, claims)
+	token.Header["kid"] = kid
+
+	s, err := token.SignedString(privateKey)
+
+	t.Logf("Token: %v %v", s, err)
+
+	return s
+}
+
+func TestToken(t *testing.T) {
+	s := createToken(t)
+
+	parsedToken, err := jwt.Parse(s, func(token *jwt.Token) (interface{}, error) {
+		return &privateKey.PublicKey, nil
+	})
+
+	if err != nil {
+		t.Errorf("Failed to json decode: %v", err)
+		return
+	}
+
+	err = parsedToken.Claims.Valid()
+
+	if err != nil {
+		t.Errorf("Failed token: %v", err)
+		return
+	}
+
+}
+
+func TestJWTValidation(t *testing.T) {
```

Comment:
This tests various tokens. There should be another test set that covers different key failure scenarios like permanent failure to load keys or key refresh failure